pkg/registry/apis/dashboard/datasources.go

```diff
@@ -0,0 +1,37 @@
+package dashboard
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/apps/dashboard/pkg/migration/schemaversion"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type datasourceInfoProvider struct {
+	datasourceService datasources.DataSourceService
+	cfg               *setting.Cfg
+}
+
+func (d *datasourceInfoProvider) GetDataSourceInfo() []schemaversion.DataSourceInfo {
+	query := datasources.GetDataSourcesQuery{DataSourceLimit: d.cfg.DataSourceLimit}
```

Comment:
What's the reasoning behind the `DataSourceLimit`?

Reply:
@toddtreece can you help me here?

Reply:
I guess if we remove the limit here, it will retrieve all data sources from the database.

What are the risks of doing that?

We probably only need:
- The data sources that are referenced in the dashboard by name or UID
- The default data source

We could go through the entire dashboard, identify the used data sources, filter that query by name, UID, or default, and once we have the data, we could rewrite the dashboard. The challenge here is that we need to review the dashboard multiple times, and some dashboards are quite large.

Reply:
@radiohead @stephaniehingtgen I removed the use of the config limit.

I see there is a query to get "all datasources" without a limit. I pushed a commit using that query instead. Can you confirm that this is correct? I'm not familiar with that at all :)